### PR TITLE
 #161808804 Create endpoint to get a single article

### PR DESCRIPTION
**Pull request by PriscillaSam**
#### What does this PR do?
Create endpoint to get a single article. **GET** `/api/v1/articles/:slug`.

#### Description of Task to be completed
- add function to get an article with Author's information.
- add the route to get an article.

#### How should this be manually tested?
- on the terminal run `git clone https://github.com/andela/haven-ah-backend.git`.
- run `git checkout ft-get-article-161808804`.
- ensure that PostgreSQL is running on your local machine.
- set up databases (test and development) following the `.env.sample` file.
- run `npm instal`l to install the npm packages.
- run `npm test` to run the unit and integration tests.
To test on postman

- run npm run start-dev and open up postman.
- post an article 
signup (`localhost:5000/api/v1/users/signup`) and use email and password to log in (`localhost:5000/api/v1/users/signin`). Copy token and pass to the x-access-token header to post an article.

Sample request body to post an article
```
{
  title: "string",
  description: "string",
  body: "string"
}
```
from the response get the slug and send a get request to `/api/v1/articles/:slug`.

Sample Responses
If the article does not exist
```
{
  status: 404,
  message: "This article was not found."
}
```

What are the relevant pivotal tracker stories?
#161808804

#### Screenshots if applicable
screen shot 2018-11-09 at 6 01 33 am

### CODE REVIEW

**Comment by TheoOkafor** [comment]
Looks good @PriscillaSam. Nice job.

**Comment by ucheg6** [comment]
Fantastic work Priscilla.

File: server/repository/articleRepository.js [hunk 1]
```diff
@@ -87,6 +87,29 @@ class ArticleRepository {
...
+   * Find an article by the slug and include user information
+   * @param {string} slug Article slug to search by
+   * @returns {object} Article or null if article is not found
+   */
+  static async getSingleArticle(slug) {
+    const article = await Articles.findOne({
+      where: { slug },
+      include: [{
+        association: 'Author',
+        attributes: ['id', 'firstName', 'lastName', 'email', 'username', 'imageUrl', 'bio']
```
Comment (seyi-adeleke) [inline server/repository/articleRepository.js]: Do we really need all of this user information?

**Review by seyi-adeleke — APPROVED** [review]

File: server/repository/articleRepository.js (hunk 1 above)
Comment (PriscillaSam) [reply server/repository/articleRepository.js]: @seyi-adeleke would remove unnecessary information.

**Pull request merged**